app/services/schedule_generator.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+# It generates a planning for the next month and enroll the members with have a static slot
+class ScheduleGenerator < ApplicationService
+  def initialize
+    @errors = []
+  end
+
+  def generate_schedule
+    valid = true
+    Mission.transaction do
+      generate_missions_for_next_month
+      if @errors.any?
+        valid = false
+        raise ActiveRecord::Rollback
+      end
+    end
+    valid
+  end
+
+  def generate_missions_for_next_month
+    current_day = (DateTime.current + 1.month).at_beginning_of_month
+    next_month = current_day.month
+    while current_day.month == next_month
+      generate_missions_for_a_day(current_day)
+      current_day += 1.day
+    end
+  end
+
+  def generate_missions_for_a_day(current_day)
+    return if current_day.strftime('%a') == 'Sun'
+
+    generate_current_hours(current_day).each do |current_hour|
+      mission = FactoryBot.create :mission, name: 'permanence clac', description: 'permanence clac',
+                                            start_date: current_hour, min_member_count: 2
```

Comment:
Do not use FactoryBot for production code. Rather be explicit about what attributes are being set. Otherwise, fiddling with our test suite could have unwanted impact on production.
We only use FactoryBot for tests and seeds.